### DISCUSSION TITLE
Make (ts :on today) respect org-extend-today-until

### DISCUSSION THREAD

**Discussion by larrasket** [alphapapa/org-ql#404] [Ideas]
Title. Org already respects this value holistically by using the function `org-today` to get today's date. From org.el:

```elisp
(defun org-today ()
  "Return today date, considering `org-extend-today-until'."
  (time-to-days
   (time-since (* 3600 org-extend-today-until))))
```

**Comment by alphapapa** [comment]
I have mixed feelings about this idea.  On one hand, it seems like a nice way to provide compatibility and flexibility.  OTOH I generally want to avoid adding support for options like this, because, as seen in org-agenda.el, they complicate the code--over time, severely.

But, IIUC, this could already be done without any changes to `org-ql` by using a function like this:

```elisp
(defun ts-end-of-today-extended ()
  "Return timestamp at the end of today extended by `org-extend-today-until'."
  (let ((start-of-tomorrow
         (ts-apply :hour 0 :minute 0 :second 0
                   (ts-adjust 'day 1 (ts-now)))))
    (ts-adjust 'second org-extend-today-until start-of-tomorrow)))
```

Then you could use its value in a query expression, like:
```
`(ts :to ,(ts-end-of-today-extended))
```

What do you think?
